src/giskardpy/qp/qp_solver_qpalm.py

```diff
@@ -89,9 +111,19 @@ def __init__(self, weights: cas.Expression, g: cas.Expression, lb: cas.Expressio
 
     @profile
     def evaluate_functions(self, substitutions: np.ndarray):
-        self.weights, self.lb, self.bE, self.lbA, self.lb_bE_lbA = self.w_lb_bE_lbA_f.fast_call(substitutions)
-        self.ub, _, self.ubA, self.ub_bE_ubA = self.ub_bE_ubA_f.fast_call(substitutions)
-        self.g = np.zeros(self.weights.shape)
+        if self.use_manipulability:
+            self.calc_det_of_jjt_manipulability(substitutions)
+            self.weights, self.lb, self.bE, self.lbA, self.ub, _, self.ubA, m_grad, m, self.lb_bE_lbA, self.ub_bE_ubA = self.combined_vector_f.fast_call(
+                np.append(substitutions, self.det))
+            self.g = np.zeros(self.weights.shape)
+            self.g[:len(m_grad)] = m_grad
+            god_map.qp_controller.manipulability_indexes[1] = god_map.qp_controller.manipulability_indexes[0]
+            god_map.qp_controller.manipulability_indexes[0] = m
+        else:
+            self.weights, self.lb, self.bE, self.lbA, self.ub, _, self.ubA, self.lb_bE_lbA, self.ub_bE_ubA = self.combined_vector_f.fast_call(
```

Comment:
yeah, here the second bE just gets thrown away

Reply:
It seems that the second bE is necessary to resolve self.ub_bE_ubA with a additional view from the StackedCompiledFunction

Reply:
true